esp-hal/src/rmt.rs

```diff
@@ -969,12 +1017,18 @@ mod impl_for_chip {
 
 /// RMT Channel
 #[non_exhaustive]
-#[derive(Debug)]
 pub struct Channel<M, const CHANNEL: u8>
 where
     M: crate::Mode,
 {
     phantom: PhantomData<M>,
+    _guard: PeripheralGuard,
+}
+
+impl<M: crate::Mode, const CHANNEL: u8> core::fmt::Debug for Channel<M, CHANNEL> {
```

Comment:
ah yeah - having `Debug` on the `PeripheralGuard` is actually better 👍 

I think in many places we unconditionally implement Debug to allow `unwrap` - not having it would (conditionally) would be a breaking change (which we might or might not want)